src/lib/api/group/client/groupApiClient.client.ts

```diff
@@ -77,3 +77,34 @@ export const putRenameGroup = async (
     return Err('API response is invalid');
   }
 };
+
+/**
+ * Interface representing the API response for the function to generate a invitation link
+ * @property invitationLinkHash - string, a hash for invitation link
+ */
+export interface IGenerateInvitationLink {
+  invitationLinkHash: string;
+}
+
+/**
+ * generate a invitation link hash
+ * @param groupId unique Id for generating invitation link hash
+ * @returns A hash: IGenerateInvitationLink of inivitation link
+ */
+export const generateInvitationLinkHash = async (
+  groupId: string,
+): Promise<Result<string, string>> => {
```

Comment:
Currently, it is not very clear what the type `string` refers to for the return type:
```ts
Result<string, string>
```

I suggest creating an interface for the return value and changing the return type like so:
```ts
interface IPutgenerateInvitationLinkHashResponse {
  hash: string
  // This structure allows to add more values for the future
  // It's also kind of a convention to always make the return value an object
}
:
Promise<Result<IPutgenerateInvitationLinkHashResponse, string>>
```